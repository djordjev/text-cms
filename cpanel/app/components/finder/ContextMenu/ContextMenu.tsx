import { Link } from '@remix-run/react';
import { IconFilePlus, IconFolderPlus } from '@tabler/icons-react';
import { FC } from 'react';

export interface ContextMenuProps {}

const YELLOW = '#f5c24a';
const BLUE = '#5ab1e0';

const ContextMenu: FC<ContextMenuProps> = () => {
  return (
    <ul className="u-menu u-bg-base-200 u-w-2z u-rounded-box">
      <li>
        <Link
          className="hover:u-underline u-flex u-items-center"
          to="?action=new-folder"
        >
          <IconFolderPlus color={YELLOW} />
          New Folder
        </Link>
      </li>
      <li>
        <Link
          className="hover:u-underline u-flex u-items-center"
          to="?action=new-file"
        >
          <IconFilePlus color={BLUE} />
          New File
        </Link>
      </li>
    </ul>
  );
};

export { ContextMenu };
