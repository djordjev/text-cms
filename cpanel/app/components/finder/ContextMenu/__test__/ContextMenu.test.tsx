import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { ContextMenu, ContextMenuProps } from '../ContextMenu';

test('ContextMenu', () => {
  const props: ContextMenuProps = {};

  const createComponent = () => {
    const RemixStub = createRemixStub([
      { path: '/', Component: () => <ContextMenu {...props} /> }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(screen.getByText('New File')).toBeInTheDocument();
  expect(screen.getByText('New Folder')).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
