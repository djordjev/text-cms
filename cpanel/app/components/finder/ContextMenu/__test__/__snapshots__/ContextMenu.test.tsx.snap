// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextMenu 1`] = `
NodeList [
  <ul
    class="u-menu u-bg-base-200 u-w-2z u-rounded-box"
  >
    <li>
      <a
        class="hover:u-underline u-flex u-items-center"
        href="/?new=folder"
      >
        <svg
          class="tabler-icon tabler-icon-folder-plus"
          fill="none"
          height="24"
          stroke="#f5c24a"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 19h-7a2 2 0 0 1 -2 -2v-11a2 2 0 0 1 2 -2h4l3 3h7a2 2 0 0 1 2 2v3.5"
          />
          <path
            d="M16 19h6"
          />
          <path
            d="M19 16v6"
          />
        </svg>
        New Folder
      </a>
    </li>
    <li>
      <a
        class="hover:u-underline u-flex u-items-center"
        href="/?new=file"
      >
        <svg
          class="tabler-icon tabler-icon-file-plus"
          fill="none"
          height="24"
          stroke="#5ab1e0"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 3v4a1 1 0 0 0 1 1h4"
          />
          <path
            d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z"
          />
          <path
            d="M12 11l0 6"
          />
          <path
            d="M9 14l6 0"
          />
        </svg>
        New File
      </a>
    </li>
  </ul>,
]
`;
